src/tequilapi-client.ts

```diff
@@ -227,22 +231,26 @@ export class HttpTequilapiClient implements TequilapiClient {
     await this.http.put(`identities/${id}/email`, { email })
   }
 
-  public async authChangePassword(
-    username: string,
-    oldPassword: string,
-    newPassword: string
-  ): Promise<void> {
-    return this.http.put(`auth/password`, {
-      username,
-      // eslint-disable-next-line @typescript-eslint/camelcase
-      oldPassword: oldPassword,
-      // eslint-disable-next-line @typescript-eslint/camelcase
-      newPassword: newPassword,
+  public authSetToken(token: string): void {
+    this.http.setHeaders({
+      Authorization: 'Bearer ' + token,
     })
   }
 
-  public async authLogin(username: string, password: string): Promise<void> {
-    return this.http.post(`auth/login`, { username, password })
+  public async authAuthenticate(request: AuthRequest): Promise<AuthResponse> {
+    return this.http.post(`auth/authenticate`, request)
```

Comment:
I am still not sure we need token based auth, perhaps @ignasbernotas could provide more details on hos side of things. Or we can go over this topic in general, as in have one more grooming.